commands/Tools/Step 3 - Modify schematisation/control_structures.py

```diff
@@ -0,0 +1,210 @@
+        self.dockwidget_controlled_structures.\
+            combobox_input_structure_id.clear()
+        db_key = self.dockwidget_controlled_structures.combobox_input_model\
+            .currentText()  # name of database
+        db_entry = self.databases[db_key]
+
+        _db_settings = db_entry['db_settings']
+
+        if db_entry['db_type'] == 'spatialite':
+            host = _db_settings['db_path']
+            db_settings = {
+                'host': host,
+                'port': '',
+                'name': '',
+                'username': '',
+                'password': '',
+                'schema': '',
+                'database': '',
+                'db_path': host,
+            }
+        else:
+            db_settings = _db_settings
+            db_settings['schema'] = 'public'
+        control_structure = ControlledStructures(flavor=db_entry['db_type'])
+        control_structure.start_sqalchemy_engine(db_settings)
+        # Get all id's of the connection nodes
+        with control_structure.engine.connect() as con:
+            rs = con.execute(
+                '''SELECT id FROM v2_connection_nodes;'''
+            )
+            connection_node_ids = rs.fetchall()
+        con.close()
+        for connection_node_id in connection_node_ids:
+            id_nr = connection_node_id[0]
+            self.dockwidget_controlled_structures.\
+                combobox_input_measuring_point_id.addItem(str(id_nr))
+        # Get all id's of the structures
+        control_structure.start_sqalchemy_engine(db_settings)
+        with control_structure.engine.connect() as con:
+            rs = con.execute(
+                '''SELECT weir_id FROM v2_weir_view;'''
+            )
+            weir_ids = rs.fetchall()
+        con.close()
+        for weir_id in weir_ids:
+            id_nr = weir_id[0]
+            self.dockwidget_controlled_structures.\
+                combobox_input_structure_id.addItem(str(id_nr))
+
+    def save_control_structure(self):
+        """
+        Function to save the created control in the designated table.
+        Currently only works for table control.
+        """
+        control_type = TABLE_CONTROL
+        # Get the model
+        db_key = self.dockwidget_controlled_structures\
+            .combobox_input_model.currentText()
+        db_entry = self.databases[db_key]
+
+        _db_settings = db_entry['db_settings']
+
+        if db_entry['db_type'] == 'spatialite':
+            host = _db_settings['db_path']
+            db_settings = {
+                'host': host,
+                'port': '',
+                'name': '',
+                'username': '',
+                'password': '',
+                'schema': '',
+                'database': '',
+                'db_path': host,
+            }
+        else:
+            db_settings = _db_settings
+            db_settings['schema'] = 'public'
+        control_structure = ControlledStructures(
```

Comment:
Defining the ``db_settings`` is a lot of boilerplate code that is repeated at least twice. Better to use a helper function or to put the empty dict elsewhere and then just update the section to need